### PR TITLE
Implement Gravatar QuickEditor

### PR DESCRIPTION
**Pull request by AdamGrzybkowski**
This PR implements the QuickEditor from the Gravatar SDK. It fully replaces the Avatar update flow, including uploading already existing photos or new ones from the camera.

**A few things worth noting:**
- One of Gravatar's compose dependencies bumps the foundation library. This caused the WP build to fail because a few things got deprecated. These issues were fixed in https://github.com/wordpress-mobile/WordPress-Android/commit/5b1adac3c25f43908889e1c99c3603e47a6db8d1. If you would prefer, I can update the compose BOM (only this specific artifact) and apply those changes in a separate PR.
- Gravatar uses UCrop like the WP app. We found an issue where the cropped image size (file size) is not changed. We've tried to fix that in UCrop ([PR](https://github.com/Yalantis/uCrop/pull/926)) but things aren't moving fast there, so we released the Automattic fork with the fix applied. I had to update WP to the Automattic version to not cause conflicts. 
- I've added a feature flag for the QuickEditor. Its purpose is a safety mechanism to be able to turn off the integration in case something doesn't work as it should. I'm not sure where to set this remotely or whether this has to happen now or when it's needed. Can you guide me here? 

Here's how the QuickEditor looks like in WP on the `MeFragment`

https://github.com/user-attachments/assets/f4488cf1-76d9-4e2b-815b-37dbb2528a64

And here's the `SignupEpilogueFragment`

https://github.com/user-attachments/assets/a6d5fabf-8651-47ae-a47d-16e7f8c23ec3

-----

## To Test:

1. Go through the sign-up flow
2. On the `SignupEpilogueFragment` tap the avatar to upload a new image
3. Confirm the avatar was uploaded
4. Continue and go to the `MeFragment` by tapping the `Me` option
5. Tap on the avatar
6. Confirm the QuickEditor was opened
7. Upload a new avatar, play with the QE.
8. Modify the `GRAVATAR_QUICK_EDITOR` feature flag in build.gradle to `false`
9. Repeat steps 1-7 - this should behave same as it works now without the QuickEditor.

-----

## Regression Notes

1. Potential unintended areas of impact

    - no

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>GravatarQuickEditorFeatureConfig</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AdamGrzybkowski** [comment]
I added `unit-tests-exemption` label because Danger was complaining about the lack of tests for `GravatarQuickEditorFeatureConfig`. From what I've checked, those types of classes are not tested in the WP repo.

**Comment by AdamGrzybkowski** [comment]
There seems to be some kind of Lint issue with the navigation-compose library from the QuickEditor... I will mark it as a draft until I figure it out.

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ androidx-constraintlayout-compose = '1.0.1'
 androidx-core = '1.13.1'
 androidx-fragment = '1.8.4'
 androidx-lifecycle = '2.8.5'
-androidx-navigation = '2.7.7'
+androidx-navigation = '2.8.3'
```
Comment (AdamGrzybkowski) [inline gradle/libs.versions.toml]: There's a bug in 2.8.1 (used by QuickEditor) that causes some Lint warnings. WP is configured to fail on those so I had to bump the library here to fix the issue. 

Here's the issue tracker with a bit more info - https://issuetracker.google.com/issues/368070326

We will update the library in the Gravatar SDK as well, but for now, this should do the trick. 

@nbradbury Any thoughts on this?

Comment (nbradbury) [reply gradle/libs.versions.toml]: Normally I'd opt for updating a library in a dedicated PR, but I just tested the app separately with androidx-navigation v2.8.3 and ran into no problems, so I'm fine with updating it here.

**Comment by nbradbury** [comment]
The QuickEditor is very nice! I have some feedback on it that's separate from this PR.

If I deny camera permission then try to take a photo, I see this dialog. This is fine, but it would be nicer if the dialog enabled the user to go directly to the app settings.

permission

The QuickEditor has a "Done" button at the top left which seems odd to me. A standard **X** button seems better.

done

After I tap an image to make it my profile photo, there's a two-second delay before I see the toast informing me the photo was updated. Is there a way to provide immediate feedback? Also, the wording "May take a few minutes to appear everywhere" seems clipped. Should that be "**It** may take a few minutes to appear everywhere"?


https://github.com/user-attachments/assets/674b03ec-ea33-47da-815b-4d0ddce2ca77

**Review by nbradbury — APPROVED** [review]
@AdamGrzybkowski I'm approving this PR, feel free to merge when ready but note that as with the parent PR, I'm unable to test the `SignupEpilogueFragment`.

**Comment by AdamGrzybkowski** [comment]
>If I deny camera permission then try to take a photo, I see this dialog. This is fine, but it would be nicer if the dialog enabled the user to go directly to the app settings.

That makes sense, I will see what can we do here from an SDK perspective. Issue created - https://github.com/Automattic/Gravatar-SDK-Android/issues/422

>The QuickEditor has a "Done" button at the top left which seems odd to me. A standard X button seems better.

We followed the designs, but yeah in the Android world an `x` is more common, I think. Will talk with the team. 

>After I tap an image to make it my profile photo, there's a two-second delay before I see the toast informing me the photo was updated. I see the toast informing me the photo was updated. Is there a way to provide immediate feedback?

We show the confirmation when the avatar is updated. We could assume a positive result and do that instantly, but we opted for a progress indicator instead. Have you missed it? 

>Also, the wording "May take a few minutes to appear everywhere" seems clipped. Should that be "It may take a few minutes to appear everywhere"?

Yes, thanks for catching that! Issue - https://github.com/Automattic/Gravatar-SDK-Android/issues/423

**Comment by nbradbury** [comment]
> We could assume a positive result and do that instantly, but we opted for a progress indicator instead. Have you missed it?

Oh, yes - I did miss the progress indicator. It's a bit subtle, but I confess to not having super vision so it's likely my fault I missed it :)

**Comment by AdamGrzybkowski** [comment]
>Oh, yes - I did miss the progress indicator. It's a bit subtle, but I confess to not having super vision so it's likely my fault I missed it :)

I think we can work on a better contrast there as well. Thanks for the feedback!

**Comment by AdamGrzybkowski** [comment]
@hamorillo When you have time, would you mind reviewing and testing this integration? 🙏

**Comment by hamorillo** [comment]
💡 I wonder if, after switching avatars, we should keep the `forceRefresh` enabled after every `onResume` to avoid the following case:

https://github.com/user-attachments/assets/0c5df3fc-30e2-4c7c-9a7e-550c13b575f4

Appart from that, LGTM!

**Comment by hamorillo** [comment]
Another minor detail (which probably is easy to say but more complicated to fix) is that we are not updating the avatar in the bottom bar:

image

**Comment by AdamGrzybkowski** [comment]
>Another minor detail (which probably is easy to say but more complicated to fix) is that we are not updating the avatar in the bottom bar:

I know about this, but this issue was there before. So it should be tackled separately as it's not directly tied to the QE.

**Comment by AdamGrzybkowski** [comment]
>💡 I wonder if, after switching avatars, we should keep the forceRefresh enabled after every onResume to avoid the following case:


I think we should, it's a bit annoying because the `forceRefresh` becomes the new default 🙄

**Comment by hamorillo** [comment]
> I think we should, it's a bit annoying because the forceRefresh becomes the new default 🙄

I agree it's a pain. In this case, we can only enable it after switching avatars, which won't be very common. :(

**Comment by AdamGrzybkowski** [comment]
>I agree it's a pain. In this case, we can only enable it after switching avatars, which won't be very common. :(


I think I was able to solve this without using the `forceRefresh` too often. There's this bitmap cache that is used to store the Avatar so I'm now caching a new bitmap for the avatarUrl (without the force refresh random string query part) when a new one was selected. 


@hamorillo Can you try it?

**Comment by AdamGrzybkowski** [comment]
@nbradbury Could you give me access to Firebase or tell me which tests have failed? I'm running tests locally, but those that failed seem to be unrelated to my changes so I wonder if different ones failed on CI.

**Comment by nbradbury** [comment]
> Could you give me access to Firebase or tell me which tests have failed? 

@AdamGrzybkowski I'm not sure I can give you access to Firebase. @wzieba @ParaskP7 Is this something you can do?

> I'm running tests locally, but those that failed seem to be unrelated to my changes so I wonder if different ones failed on CI.

You're right, it's the e2ee tests that are failing, which are unrelated. I'll do a rebuild in CI and see if that fixes the problem.

**Update:** Rebuilding solved the problem. Apologies for the false alarm!

**Comment by AdamGrzybkowski** [comment]
@nbradbury Thanks! I rerun the build yesterday but it failed again. I guess some external factors were in play.

**Comment by ParaskP7** [comment]
> @AdamGrzybkowski I'm not sure I can give you access to Firebase. @wzieba @ParaskP7 Is this something you can do?

Hey @AdamGrzybkowski @nbradbury , thanks for the ping, you would need to post a System Request for Firebase access (example: pMz3w-c9y-p2).

**Comment by AdamGrzybkowski** [comment]
@ParaskP7 No need for that. It seems like these were some flaky tests and I no longer need access to Firebase to check them.

**Review by hamorillo — APPROVED** [review]
@AdamGrzybkowski, I confirm it works great for me!

**Pull request merged**